### PR TITLE
Training

### PR DESCRIPTION
**Pull request by syKevinPeng**

### CODE REVIEW

**Review by raytu — APPROVED** [review]

File: add_output_layers.py [hunk 1]
```diff
@@ -13,21 +13,21 @@ def first_output_stage(backbone_layer, out_shape, out_layer_name=''):
 
     return y
 
-def add_output_layers(orientation_output_type, backbone_layer):
+def add_output_layers(orientation_output_type, backbone_layer, num_sec = 0):
```
Comment (raytu) [inline add_output_layers.py:16]: num_sectors, num_multibins

File: training.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import numpy as np
+import data_processing as dp
+import os, argparse, time, pickle
+
+# Processing argument
+parser = argparse.ArgumentParser(description='Training Model')
+parser.add_argument(dest = 'orientation', type = str, 
+                   help = 'Orientation Type of the model. Options are tricosine, alpha, rot_y, alpha_sector, rot_y_sector, multibin')
+parser.add_argument('--num_sector', dest = 'NUM_SECTOR', type = int, default = 4,
+                   help = 'Number of sector for orientation type alpha_sector, rot_y_sector, and multibin. Default value is 4')
```
Comment (raytu) [inline training.py:16]: add num_multibin

Comment (raytu) [reply training.py:16]: add data_label

Comment (raytu) [reply training.py:16]: add data_img

File: training.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                   help = 'Define the batch size for training. Default value is 8')
+parser.add_argument('--weight_dir', dest = 'weight_dir', type = str,
+                   help = 'Relative path to save weights. Default path is weights')
+parser.add_argument('--epoch', dest = 'num_epoch', type = int, default=100,
+                    help = 'Number of epoch used for training. Default value is 100')
+
+args = parser.parse_args()
+
+# data directory
+data_label = 'dataset/training/label_2/'
```
Comment (raytu) [inline training.py:27]: generate a paths with pathlib

File: training.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        raise Exception('Wrong Orientation Type. Use python training.py -h to see more')
+        
+    
+    print('Training on orientation type:[' ,args.orientation,'] with batch_size =',BATCH_SIZE, 'and num_sector is =',NUM_SECTOR)    
+    # Generator config
+    print('Loading generator')
+    generator = dp.KittiGenerator(label_dir= data_label, image_dir= data_img,sectors = NUM_SECTOR, batch_size = BATCH_SIZE)
+    # Model callback config
+    # checkpoint_path = os.path.join(weight_dir,'model.{epoch:02d}-{val_loss:.2f}.h5')
+    checkpoint_path = os.path.join(weight_dir, 'model.{epoch:02d}.h5')
```
Comment (raytu) [inline training.py:76]: save mode file with orientation type.

**Comment by syKevinPeng** [comment]
Fix multibin. Now we can train on multibin mod
add kitti_dir  command arg
add num_multibin command arg
Add num_sector and num_multibin to add_output_layers.py
modify the output layer for mutibin in add_output_layers.py

**Review by raytu — APPROVED** [review]

**Pull request merged**